pkg/agent/agent.go

```diff
@@ -726,6 +726,7 @@ func (i *Initializer) setupDefaultTunnelInterface() error {
 	if portExists {
 		if i.networkConfig.TrafficEncapMode.SupportsEncap() &&
 			tunnelIface.TunnelInterfaceConfig.Type == i.networkConfig.TunnelType &&
+			tunnelIface.TunnelInterfaceConfig.DestinationPort == int32(i.networkConfig.TunnelPort) &&
```

Comment:
I think `i.networkConfig.TunnelPort` is already int32?

Reply:
Thanks for the review, updated.